packages/transmute-framework/README.md

```diff
@@ -1 +1,166 @@
+
+
+Once you have setup the minikube cluster:
+
+```
+
+export KONG_ADMIN_URL=$(minikube service gateway-kong-admin --url | sed 's,http://,https://,g')
+export KONG_PROXY_URL=$(minikube service gateway-kong-proxy --url | sed 's,http://,https://,g')
+
+echo $KONG_PROXY_URL
+
+curl -k -X GET \
+  --url $KONG_ADMIN_URL/apis \
+  | jq -r '.'
+```
+
+You can configure your `/etc/hosts` like so:
+
+```
+192.168.99.100  transmute.minikube
+192.168.99.100  ipfs.transmute.minikube
+192.168.99.100  ganache.transmute.minikube
+```
+
+With the `/scripts/configure-hosts.sh` from the root of the repo.
+
+Update minikube in `./src/transmute-config/env.json`. 
+
+The kong proxy port is likly the only thing that will change when using minikube locally.
+
+```
+npm i
+npm run truffle:test
+npm run truffle:migrate
+npm run test
+```
+
+#### Usage
+
+The `EventStoreFactory` contract can be used to create `EventStore` contracts.
+
+```
+const eventStoreFactoryArtifact = require('../../../build/contracts/EventStoreFactory.json');
+const transmuteConfig = {
+  "keenConfig": {
+    ...
+  },
+  "ipfsConfig": {
+    "host": "ipfs.transmute.minikube",
+    "port": 32443,
+    "protocol": "https"
+  },
+  "web3Config": {
+    "providerUrl": "https://ganache.transmute.minikube:32443"
+  }
+}
+eventStoreFactory = new EventStoreFactory({
+  eventStoreFactoryArtifact,
+  ...transmuteConfig
+});
+
+await eventStoreFactory.init();
+let result = await eventStoreFactory.createEventStore(
+  accounts[0]
+);
+```
+
+Saving key-value pairs to an `EventStore` contract can be done like so:
+
+```
+const EventStore = require('../index.js');
+const transmuteConfig = require('../../transmute-config');
+const eventStoreArtifact = require('../../../build/contracts/EventStore.json');
+
+const eventStore = new EventStore({
+  eventStoreArtifact,
+  ...transmuteConfig
+});
+
+await eventStore.init();
+
+const event = {
+  key: {
+    type: 'ACCOUNT_CREATED',
+    value: '0x123'
+  },
+  value: {
+    name: 'alice'
+  }
+}
+
+let result = await eventStore.write(
+  accounts[0],
+  event.key,
+  event.value
+);
+
+```
+
+Events can be retrieved using:
+
+```
+let events = await eventStore.getSlice(0, 2);
+```
+
+##### Encryption should be used before saving sensitive information to Ethereum and IPFS!
+
+Stream models can be built from the events like so:
+
+```
+const filter = event => {
+  return true;
+};
+const reducer = (state, event) => {
+  let eventHash = eventStore.web3.sha3(JSON.stringify(event));
+  const eventHashes = new Set(state.eventHashes || []);
+  eventHashes.add(eventHash);
+  return {
+    ...state,
+    eventHashes: Array.from(eventHashes)
+  };
+};
+const streamModel = new StreamModel(eventStore, filter, reducer);
+await eventStore.write(
+  accounts[0],
+  someNewEvent.key,
+  someNewEvent.value
+);
+await streamModel.sync();
+```
+
+Stream models filter events, and then use a reducer function to build up state, just like Redux!
+
```

Comment:
Might be useful to add on links like this => https://github.com/transmute-industries/transmute/tree/master/packages/transmute-framework/src/event-store/__tests__

To quickly show the testing of this code.